File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/MessagePacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MessagePacket implements ServerPacket {
  private String sender;
  private String content;
  private String target;
  private int senderId;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_MESSAGE;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeString(stream, sender);
    writer.writeString(stream, content);
    writer.writeString(stream, target);
    writer.writeInt32(stream, senderId);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/MessagePacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.BanchoPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataWriter

class MessagePacket(
    var sender: String = "",
    var content: String = "",
    var target: String = "",
    var senderId: Int = 0,
) : BanchoPacket(), BanchoPacket.Server {
    override val type = PacketType.BANCHO_MESSAGE

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeString(out, sender)
        writer.writeString(out, content)
        writer.writeString(out, target)
        writer.writeInt32(out, senderId)
    }
}
